tools/rostopic/src/rostopic/__init__.py

```diff
@@ -173,40 +176,123 @@ def print_hz(self):
             max_delta = max(self.times)
             min_delta = min(self.times)
 
-            self.last_printed_tn = self.msg_tn
-        print("average rate: %.3f\n\tmin: %.3fs max: %.3fs std dev: %.5fs window: %s"%(rate, min_delta, max_delta, std_dev, n+1))
+            self.last_msg_tn = self.msg_tn
+
+        return rate, min_delta, max_delta, std_dev, n+1
+
+    def print_hz(self):
+        """
+        print the average publishing rate to screen
+        """
+        if not self.times:
+            return
+        ret = self.get_hz()
+        if ret is None:
+            print("no new messages")
+            return
+        rate, min_delta, max_delta, std_dev, window = ret
+        print("average rate: %.3f\n\tmin: %.3fs max: %.3fs std dev: %.5fs window: %s"%(rate, min_delta, max_delta, std_dev, window))
 
 def _sleep(duration):
     rospy.rostime.wallsleep(duration)
 
-def _rostopic_hz(topic, window_size=-1, filter_expr=None):
+
+def get_parent_topics(topic, parent_topics=None, impl=False):
+    """Get topics to which the specified topic subscribes to.
```

Comment:
Conceptionally a topic does not subscribe to another topic. Therefore I think the current naming is problematic. I don't have a good name for it though. But the docblock need to clarify what it means.